service/src/main/java/org/apache/celeborn/server/common/service/config/FsConfigServiceImpl.java

```diff
@@ -88,10 +85,12 @@ private synchronized void refresh() {
       }
     } catch (Exception e) {
       LOG.warn("Refresh dynamic config error: {}", e.getMessage(), e);
+      return;
     }
 
     tenantConfigAtomicReference.set(tenantConfs);
-    systemConfigAtomicReference.set(systemConfig == null ? new SystemConfig(celebornConf) : systemConfig);
+    systemConfigAtomicReference.set(
+        systemConfig == null ? new SystemConfig(celebornConf) : systemConfig);
```

Comment:
why not initialize the systemConfig on defining, then we can also `null` propagation

Reply:
Actually, the current design treats CelebornConf as the baseConf for SystemConfig, so that the interfaces of ConfigServer can be unified to DynamicConf.  Of course, it's also possible to have SystemConfig as null, but that would require ConfigService to implement other interfaces such as getSystemValue/getTenantValue. However, this may not easy to make a configuration cache. Therefore, I adopted the first design.